Yoast/wordpress-seo
# Convert general sitemap debug flag to specific one

Instead of checking for WP_DEBUG or WPSEO_DEBUG in combination with WP_DEBUG_DISPLAY we want to be able to more specifically output debug information on the sitemap pages.

## Summary

This PR can be summarized in the following changelog entry:

* Decoupled sitemap debug information from general `WP_DEBUG` development flag and introduced `YOAST_SEO_DEBUG_SITEMAPS` to better control this functionality.

## Relevant technical choices:

* Moved the functionality to its own method

## Test instructions

This PR can be tested by following these steps:

* Configure `WP_DEBUG` and `WP_DEBUG_DISPLAY` to be `true`
* Or use `define( 'WPSEO_DEBUG', true );`
* Open a sitemap specific page (post/page)
* Inspect the source of the page
* See information being output regarding memory usage and number of queries used
* Check out this PR
* See the information not being output
* Add `define( 'YOAST_SEO_DEBUG_SITEMAPS', true );` to your `wp-config.php` or `wp-seo.php`
* See the debug information return

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo/issues/12082
